### PR TITLE
[Card] add support for destructive footer and section actions

### PR DESCRIPTION
**Pull request by chloerice**
### WHY are these changes introduced?

Follow up to https://github.com/Shopify/polaris-react/pull/1603

### WHAT is this pull request doing?

This PR adds support for destructive footer and section actions. It does not add that support to the header actions formed by `Card.actions` or `Card.Header.actions` because destructive actions should not be taken before reviewing the card or card section.

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

Check out the new destructible examples "Card with destructive footer action" and "Card section with destructive action" in the latest Storybook deployment 

*or*

<details>
<summary><code>yarn dev</code> and copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import * as React from 'react';
import {Page, Layout, Card, List} from '../src';

interface State {}

export default class Playground extends React.Component<{}, State> {
  render() {
    return (
      <Page title="Playground">
        <Layout>
          <Layout.Section>
            <Card
              title="Shipment 1234"
              secondaryFooterAction={{
                content: 'Cancel shipment',
                destructive: true,
              }}
              primaryFooterAction={{content: 'Add tracking number'}}
            >
              <Card.Section title="Items">
                <List>
                  <List.Item>1 × Isis Glass, 4-Pack</List.Item>
                  <List.Item>1 × Anubis Cup, 2-Pack</List.Item>
                </List>
              </Card.Section>
            </Card>
            <Card title="Customer">
              <Card.Section>
                <p>John Smith</p>
              </Card.Section>
              <Card.Section
                title="Contact Information"
                actions={[
                  {content: 'Unsubscribe', destructive: true},
                  {content: 'Edit'},
                ]}
              >
                <p>john.smith@example.com</p>
              </Card.Section>
            </Card>
          </Layout.Section>
        </Layout>
      </Page>
    );
  }
}
```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [x] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [x] Updated the component's `README.md` with documentation changes
* [x] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: src/components/Card/Card.tsx [hunk 1]
```diff
@@ -20,9 +24,9 @@ export interface Props {
   /** Card header actions */
   actions?: DisableableAction[];
   /** Primary action in the card footer */
-  primaryFooterAction?: Action;
+  primaryFooterAction?: DestructableAction;
   /** Secondary action in the card footer */
-  secondaryFooterAction?: Action;
+  secondaryFooterAction?: DestructableAction;
```
Comment (qq99) [inline src/components/Card/Card.tsx]: As we discussed in Slack, I think we should do `ComplexAction` for both

File: UNRELEASED.md [hunk 2]
```diff
@@ -24,6 +24,7 @@ Use [the changelog guidelines](https://git.io/polaris-changelog-guidelines) to f
 - Removed transition on tag button hover state [#1337](https://github.com/Shopify/polaris-react/pull/1337)
 - Added `textAlign` prop to Button ([#1576](https://github.com/Shopify/polaris-react/pull/1576))
 - Made `Button` red when given both the `plain` and `destructive` props ([#1603](https://github.com/Shopify/polaris-react/pull/1603))
+- Added support for destructive actions in `Card` and `Card.Section` ([#1622](https://github.com/Shopify/polaris-react/1622))
```
Comment (qq99) [inline UNRELEASED.md]:
```suggestion
- Added support for complex actions in `Card` and `Card.Section` ([#1622](https://github.com/Shopify/polaris-react/1622))
```

Comment (qq99) [reply UNRELEASED.md]: Not really sure how to describe the actions lol, but they're more than just destructive now

Comment (chloerice) [reply UNRELEASED.md]: Done!

**Pull request merged**